### PR TITLE
Stn Bid Adapter: initial release

### PR DESCRIPTION
**Pull request by lasloche**
## Type of change

- [X] New bidder adapter   

## Description of change

Add New STN Bid Adapter

### CODE REVIEW

**Comment by lasloche** [comment]
Hi @ncolletti, can you, please, give me an update on when this PR is going to be reviewed? Thanks!

File: modules/stnBidAdapter.md [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+                context: 'instream'
+            }
+        },
+        bids: [{
+            bidder: 'stn',
+            params: {
+                org: '0987654321abcdef87654321', // Required
+                floorPrice: 2.00, // Optional
+                placementId: 'video-test', // Optional
+                testMode: false // Optional
```
Comment (ncolletti) [inline modules/stnBidAdapter.md:50]: Please update your endpoint to return valid bids with these params

File: modules/stnBidAdapter.js [hunk 2]
```diff
@@ -0,0 +1,488 @@
...
+        bidResponses.push(bidResponse);
+      });
+    }
+
+    return bidResponses;
+  },
+  getUserSyncs: function (syncOptions, serverResponses) {
+    const syncs = [];
+    for (const response of serverResponses) {
+      if (syncOptions.iframeEnabled && response.body.params.userSyncURL) {
```
Comment (ncolletti) [inline modules/stnBidAdapter.js:106]: Update to use `deepAccess` for `response.body.params.userSyncURL` and `response.body.params.userSyncPixels`

File: modules/stnBidAdapter.js [hunk 3]
```diff
@@ -0,0 +1,488 @@
...
+function getSupplyChain(schainObject) {
+  if (isEmpty(schainObject)) {
+    return '';
+  }
+  let scStr = `${schainObject.ver},${schainObject.complete}`;
+  schainObject.nodes.forEach((node) => {
+    scStr += '!';
+    scStr += `${getEncodedValIfNotEmpty(node.asi)},`;
+    scStr += `${getEncodedValIfNotEmpty(node.sid)},`;
+    scStr += `${node.hp ? encodeURIComponent(node.hp) : ''},`;
```
Comment (ncolletti) [inline modules/stnBidAdapter.js:189]: Should `node.hp` also use `getEncodedValIfNotEmpty`?

Comment (lasloche) [reply modules/stnBidAdapter.js]: Thanks for noticing, updated it!

File: modules/stnBidAdapter.js (hunk 2 above)
Comment (lasloche) [reply modules/stnBidAdapter.js]: Thanks for noticing, changed the lines to use `deepAccess` instead

File: modules/stnBidAdapter.md (hunk 1 above)
Comment (lasloche) [reply modules/stnBidAdapter.md:50]: Setting the `testMode` param to `true` inside the bidder params enables the test ads

**Comment by lasloche** [comment]
@ncolletti, I've replied to your comments and updated the code. Please, let me know if you have additional questions!

File: modules/stnBidAdapter.md (hunk 1 above)
Comment (ncolletti) [reply modules/stnBidAdapter.md:50]: I am seeing this response: `{"params":{"sessionId":"141c6e86e34c45"}}`. 

Could you look into what is necessary to return a valid bid response? As well, let's update your markdown file to set `testMode` to `true` with a comment so others know how to test

Comment (lasloche) [reply modules/stnBidAdapter.md]: Hi @ncolletti, I've updated the test mode params in the documentation. Can you, please, verify if it works on your end now?

**Pull request merged**